src/hipoc/hipoc_program.cpp

```diff
@@ -323,17 +323,20 @@ void HIPOCProgramImpl::BuildCodeObject(std::string params,
 
     if(miopen::EndsWith(filename, ".cpp"))
     {
+        // WORKAROUND: TODO: -Wno-unused-parameter -Wno-extra-semi-stmt bypasses issues with
+        // hipRTC_header.h
 #if MIOPEN_BUILD_DEV
-        params += " -Werror" + HipKernelWarningsString();
+        params +=
+            " -Werror -Wno-unused-parameter -Wno-extra-semi-stmt " + HipKernelWarningsString();
 #else
         params += " -Wno-everything";
 #endif
     }
     else if(miopen::EndsWith(filename, ".cl"))
     {
 #if MIOPEN_BUILD_DEV
-        params +=
-            " -Werror" + (is_kernel_str ? MiopengemmWarningsString() : OclKernelWarningsString());
+        params += " -Werror -Wno-unused-parameter -Wno-extra-semi-stmt " +
+                  (is_kernel_str ? MiopengemmWarningsString() : OclKernelWarningsString());
```

Comment:
hiprtc header problems won't affect opencl builds. Please revert, thanks.

Reply:
This is reverted :)